iOS: click event doesn't get dispatched when there is a click event listener on ShadowRoot

rniwa opened the pull request:
#### b01e40564c2149ac330006002babf439d1ecded4
<pre>
iOS: click event doesn&apos;t get dispatched when there is a click event listener on ShadowRoot
<a href="https://bugs.webkit.org/show_bug.cgi?id=218923">https://bugs.webkit.org/show_bug.cgi?id=218923</a>

Reviewed by Wenson Hsieh.

The bug was caused by WebPage::commitPotentialTap falsely treating ShadowRoot as an invalid tap target
because it doesn&apos;t have a renderer. Fixed the bug by checking the shadow host&apos;s renderer instead
much like <a href="http://commits.webkit.org/r246404">http://commits.webkit.org/r246404</a> for display: contents.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::commitPotentialTap): Fixed the bug.

* LayoutTests/fast/shadow-dom/click-eventlistener-on-shadow-root-expected.txt: Added.
* LayoutTests/fast/shadow-dom/click-eventlistener-on-shadow-root.html: Added.

Canonical link: <a href="https://commits.webkit.org/252107@main">https://commits.webkit.org/252107@main</a>
</pre>

whsieh reviewed (approved):
LGTM.

Perhaps we should rename the test from `click-eventlistener-on-shadow-root-ios.html` to just `click-eventlistener-on-shadow-root.html`, if it's meant to be run on all platforms?

rniwa committed:
iOS: click event doesn't get dispatched when there is a click event listener on ShadowRoot
https://bugs.webkit.org/show_bug.cgi?id=218923

Reviewed by Wenson Hsieh.

The bug was caused by WebPage::commitPotentialTap falsely treating ShadowRoot as an invalid tap target
because it doesn't have a renderer. Fixed the bug by checking the shadow host's renderer instead
much like http://commits.webkit.org/r246404 for display: contents.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::commitPotentialTap): Fixed the bug.

* LayoutTests/fast/shadow-dom/click-eventlistener-on-shadow-root-expected.txt: Added.
* LayoutTests/fast/shadow-dom/click-eventlistener-on-shadow-root.html: Added.

Canonical link: https://commits.webkit.org/252107@main
